From bdaa99630113b7cb9d5e564a5777497d7530b490 Mon Sep 17 00:00:00 2001
From: VladimirMangos <vladimir@getmangos.com>
Date: Tue, 15 Feb 2011 02:15:26 +0100
Subject: [PATCH] ScriptDev2 patch

---
 src/bindings/CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/bindings/CMakeLists.txt b/src/bindings/CMakeLists.txt
index fc13e3a..592d773 100644
--- a/src/bindings/CMakeLists.txt
+++ b/src/bindings/CMakeLists.txt
@@ -16,4 +16,4 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
-# add_subdirectory(universal)
+add_subdirectory(ScriptDev2)
-- 
1.7.3.1.msysgit.0

