Index: src/game/ChatHandler.cpp
===================================================================
--- src/game/ChatHandler.cpp	(revision 2927)
+++ src/game/ChatHandler.cpp	(working copy)
@@ -30,6 +30,7 @@
 #include "Guild.h"
 #include "MapManager.h"
 #include "ObjectAccessor.h"
+#include "ScriptCalls.h"
 
 void WorldSession::HandleMessagechatOpcode( WorldPacket & recv_data )
 {
@@ -356,6 +357,12 @@
 
         WPAssert(data.size() == 20 + namlen);
         GetPlayer()->SendMessageToSet( &data, true );
+
+#ifdef SCRIPT_EXTENDED
+        //Send scripted event call
+        if (pCreature && Script)
+            Script->ReciveEmote(GetPlayer(),pCreature,emote_anim);
+#endif
     }
 }
 
Index: src/game/ScriptCalls.cpp
===================================================================
--- src/game/ScriptCalls.cpp	(revision 2927)
+++ src/game/ScriptCalls.cpp	(working copy)
@@ -59,7 +59,10 @@
         ||!(testScript->ItemQuestAccept     =(scriptCallItemQuestAccept     )MANGOS_GET_PROC_ADDR(testScript->hScriptsLib,"ItemQuestAccept"     ))
         ||!(testScript->GOQuestAccept       =(scriptCallGOQuestAccept       )MANGOS_GET_PROC_ADDR(testScript->hScriptsLib,"GOQuestAccept"       ))
         ||!(testScript->GetAI               =(scriptCallGetAI               )MANGOS_GET_PROC_ADDR(testScript->hScriptsLib,"GetAI"               ))
-
+#ifdef SCRIPT_EXTENDED
+        ||!(testScript->DBCPointers         =(scriptCallDBCPointers         )MANGOS_GET_PROC_ADDR(testScript->hScriptsLib,"DBCPointers"         ))
+        ||!(testScript->ReciveEmote         =(scriptCallReciveEmote         )MANGOS_GET_PROC_ADDR(testScript->hScriptsLib,"ReciveEmote"         ))
+#endif
         )
     {
         MANGOS_CLOSE_LIBRARY(testScript->hScriptsLib);
@@ -82,6 +85,11 @@
 
     Script->ScriptsInit();
 
+#ifdef SCRIPT_EXTENDED
+    //Send DBC Store pointers
+    Script->DBCPointers(&sSpellRangeStore,&sSpellStore);
+
+#endif
     return true;
 
 }
Index: src/game/ScriptCalls.h
===================================================================
--- src/game/ScriptCalls.h	(revision 2927)
+++ src/game/ScriptCalls.h	(working copy)
@@ -49,6 +49,11 @@
 
 typedef CreatureAI* (MANGOS_IMPORT * scriptCallGetAI)  ( Creature *_Creature );
 
+#ifdef SCRIPT_EXTENDED
+typedef void(MANGOS_IMPORT * scriptCallDBCPointers) ( DBCStorage <SpellRangeEntry> *sSpellRangeStore, DBCStorage <SpellEntry> *sSpellStore);
+typedef bool(MANGOS_IMPORT * scriptCallReciveEmote) ( Player *player, Creature *_Creature, uint32 emote );
+#endif
+
 typedef struct
 {
     scriptCallScriptsInit ScriptsInit;
@@ -70,6 +75,11 @@
     scriptCallGOQuestAccept GOQuestAccept;
     scriptCallGetAI GetAI;
 
+#ifdef SCRIPT_EXTENDED
+    scriptCallDBCPointers DBCPointers;
+    scriptCallReciveEmote ReciveEmote;
+#endif
+
     MANGOS_LIBRARY_HANDLE hScriptsLib;
 }_ScriptSet,*ScriptsSet;
 
Index: src/shared/Common.h
===================================================================
--- src/shared/Common.h	(revision 2927)
+++ src/shared/Common.h	(working copy)
@@ -135,4 +135,8 @@
 #else
 #  define MANGOS_DLL_SPEC
 #endif
+
+//Extended script
+#define SCRIPT_EXTENDED
+
 #endif
