Index: include/precompiled.h
===================================================================
--- include/precompiled.h	(Revision 1910)
+++ include/precompiled.h	(Arbeitskopie)
@@ -11,6 +11,11 @@
 #include "sc_grid_searchers.h"
 #include "sc_instance.h"
 
+enum backports
+{
+    UNIT_VIRTUAL_ITEM_SLOT_ID = UNIT_VIRTUAL_ITEM_SLOT_DISPLAY
+};
+
 #ifdef WIN32
 #  include <windows.h>
     BOOL APIENTRY DllMain(HANDLE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
Index: scripts/eastern_kingdoms/shadowfang_keep/shadowfang_keep.cpp
===================================================================
--- scripts/eastern_kingdoms/shadowfang_keep/shadowfang_keep.cpp	(Revision 1910)
+++ scripts/eastern_kingdoms/shadowfang_keep/shadowfang_keep.cpp	(Arbeitskopie)
@@ -91,10 +91,10 @@
                 else
                     DoScriptText(EMOTE_UNLOCK_DOOR_AD, m_creature);
                 break;
-            case 12:
-                if (m_uiNpcEntry != NPC_ASH)
-                    m_creature->HandleEmote(EMOTE_ONESHOT_USESTANDING);
-                break;
+//            case 12:
+//                if (m_uiNpcEntry != NPC_ASH)
+//                    m_creature->HandleEmote(EMOTE_ONESHOT_USESTANDING);
+//                break;
             case 13:
                 if (m_uiNpcEntry == NPC_ASH)
                     DoScriptText(SAY_POST_DOOR_AS, m_creature);
Index: scripts/kalimdor/azuremyst_isle.cpp
===================================================================
--- scripts/kalimdor/azuremyst_isle.cpp	(Revision 1910)
+++ scripts/kalimdor/azuremyst_isle.cpp	(Arbeitskopie)
@@ -103,7 +103,7 @@
 
     void SpellHit(Unit* pCaster, const SpellEntry* pSpell)
     {
-        if (pSpell->SpellFamilyFlags2 & 0x080000000)
+        if (pSpell->Id == 28880)
         {
             m_creature->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PVP_ATTACKABLE);
             m_creature->SetStandState(UNIT_STAND_STATE_STAND);
Index: scripts/kalimdor/onyxias_lair/boss_onyxia.cpp
===================================================================
--- scripts/kalimdor/onyxias_lair/boss_onyxia.cpp	(Revision 1910)
+++ scripts/kalimdor/onyxias_lair/boss_onyxia.cpp	(Arbeitskopie)
@@ -299,7 +299,7 @@
                     DoScriptText(SAY_PHASE_2_TRANS, m_creature);
 
                     // sort of a hack, it is unclear how this really work but the values appear to be valid
-                    m_creature->SetByteValue(UNIT_FIELD_BYTES_1, 3, UNIT_BYTE1_FLAG_ALWAYS_STAND | UNIT_BYTE1_FLAG_UNK_2);
+                    m_creature->SetByteValue(UNIT_FIELD_BYTES_1, 3, UNIT_BYTE1_FLAG_ALWAYS_STAND/* | UNIT_BYTE1_FLAG_UNK_2*/);
                     m_creature->AddSplineFlag(SPLINEFLAG_FLYING);
 
                     if (m_pPointData)
Index: scripts/northrend/dragonblight.cpp
===================================================================
--- scripts/northrend/dragonblight.cpp	(Revision 1910)
+++ scripts/northrend/dragonblight.cpp	(Arbeitskopie)
@@ -101,7 +101,7 @@
     if (uiAction == GOSSIP_ACTION_INFO_DEF+1)
     {
         pPlayer->CLOSE_GOSSIP_MENU();
-        pPlayer->SendMovieStart(MOVIE_ID_GATES);
+//        pPlayer->SendMovieStart(MOVIE_ID_GATES);
     }
 
     return true;
Index: scripts/outland/auchindoun/auchenai_crypts/boss_exarch_maladaar.cpp
===================================================================
--- scripts/outland/auchindoun/auchenai_crypts/boss_exarch_maladaar.cpp	(Revision 1910)
+++ scripts/outland/auchindoun/auchenai_crypts/boss_exarch_maladaar.cpp	(Arbeitskopie)
@@ -107,10 +107,10 @@
                     DoCastSpellIfCan(m_creature->getVictim(), SPELL_MOONFIRE);
                     m_uiSpellTimer = 10000;
                     break;
-                case CLASS_DEATH_KNIGHT:
-                    DoCastSpellIfCan(m_creature->getVictim(), SPELL_PLAGUE_STRIKE);
-                    m_uiSpellTimer = 10000;
-                    break;
+//                case CLASS_DEATH_KNIGHT:
+//                    DoCastSpellIfCan(m_creature->getVictim(), SPELL_PLAGUE_STRIKE);
+//                    m_uiSpellTimer = 10000;
+//                    break;
             }
         }
         else
Index: scripts/world/item_scripts.cpp
===================================================================
--- scripts/world/item_scripts.cpp	(Revision 1910)
+++ scripts/world/item_scripts.cpp	(Arbeitskopie)
@@ -47,7 +47,7 @@
     pPlayer->SendEquipError(EQUIP_ERR_NONE, pItem, NULL);
 
     if (const SpellEntry* pSpellInfo = GetSpellStore()->LookupEntry(SPELL_ARCANE_CHARGES))
-        Spell::SendCastResult(pPlayer, pSpellInfo, 1, SPELL_FAILED_NOT_ON_GROUND);
+        Spell::SendCastResult(pPlayer, pSpellInfo, 1, SPELL_FAILED_ERROR);
 
     return true;
 }
Index: system/ScriptLoader.cpp
===================================================================
--- system/ScriptLoader.cpp	(Revision 1910)
+++ system/ScriptLoader.cpp	(Arbeitskopie)
@@ -102,7 +102,22 @@
 extern void AddSC_boss_ragnaros();
 extern void AddSC_instance_molten_core();
 extern void AddSC_molten_core();
-extern void AddSC_ebon_hold();                              // scarlet_enclave
+extern void AddSC_boss_anubrekhan();                        // naxxramas
+extern void AddSC_boss_four_horsemen();
+extern void AddSC_boss_faerlina();
+extern void AddSC_boss_gluth();
+extern void AddSC_boss_gothik();
+extern void AddSC_boss_grobbulus();
+extern void AddSC_boss_kelthuzad();
+extern void AddSC_boss_loatheb();
+extern void AddSC_boss_maexxna();
+extern void AddSC_boss_noth();
+extern void AddSC_boss_heigan();
+extern void AddSC_boss_patchwerk();
+extern void AddSC_boss_razuvious();
+extern void AddSC_boss_sapphiron();
+extern void AddSC_boss_thaddius();
+extern void AddSC_instance_naxxramas();
 extern void AddSC_boss_arcanist_doan();                     // scarlet_monastery
 extern void AddSC_boss_azshir_the_sleepless();
 extern void AddSC_boss_bloodmage_thalnos();
@@ -221,8 +236,6 @@
 extern void AddSC_boss_lieutenant_drake();
 extern void AddSC_instance_old_hillsbrad();
 extern void AddSC_old_hillsbrad();
-extern void AddSC_culling_of_stratholme();                  // COT, culling_of_stratholme
-extern void AddSC_instance_culling_of_stratholme();
 extern void AddSC_dire_maul();                              // dire_maul
 extern void AddSC_instance_dire_maul();
 extern void AddSC_boss_celebras_the_cursed();               // maraudon
@@ -277,145 +290,8 @@
 extern void AddSC_winterspring();
 
 // northrend
-extern void AddSC_boss_jedoga();                            // azjol-nerub, ahnkahet
-extern void AddSC_boss_nadox();
-extern void AddSC_boss_taldaram();
-extern void AddSC_boss_volazj();
-extern void AddSC_instance_ahnkahet();
-extern void AddSC_boss_anubarak();                          // azjol-nerub, azjol-nerub
-extern void AddSC_boss_hadronox();
-extern void AddSC_boss_krikthir();
-extern void AddSC_instance_azjol_nerub();
-extern void AddSC_trial_of_the_champion();                  // CC, trial_of_the_champion
-extern void AddSC_boss_grand_champions();
-extern void AddSC_instance_trial_of_the_champion();
-extern void AddSC_boss_anubarak_trial();                    // CC, trial_of_the_crusader
-extern void AddSC_boss_faction_champions();
-extern void AddSC_boss_jaraxxus();
-extern void AddSC_instance_trial_of_the_crusader();
-extern void AddSC_northrend_beasts();
-extern void AddSC_trial_of_the_crusader();
-extern void AddSC_twin_valkyr();
-extern void AddSC_boss_novos();                             // draktharon_keep
-extern void AddSC_boss_tharonja();
-extern void AddSC_boss_trollgore();
-extern void AddSC_instance_draktharon_keep();
-extern void AddSC_boss_colossus();                          // gundrak
-extern void AddSC_boss_eck();
-extern void AddSC_boss_galdarah();
-extern void AddSC_boss_moorabi();
-extern void AddSC_boss_sladran();
-extern void AddSC_instance_gundrak();
-extern void AddSC_boss_bronjahm();                          // ICC, forge_of_souls
-extern void AddSC_boss_devourer_of_souls();
-extern void AddSC_instance_forge_of_souls();
-extern void AddSC_boss_falric();                            // ICC, halls_of_reflection
-extern void AddSC_boss_lich_king();
-extern void AddSC_boss_marwyn();
-extern void AddSC_halls_of_reflection();
-extern void AddSC_instance_halls_of_reflection();
-extern void AddSC_boss_gafrost();                           // ICC, pit_of_saron
-extern void AddSC_boss_krick_and_ick();
-extern void AddSC_boss_tyrannus();
-extern void AddSC_instance_pit_of_saron();
-extern void AddSC_pit_of_saron();
-extern void AddSC_blood_prince_council();                   // ICC, icecrown_citadel
-extern void AddSC_boss_blood_queen_lanathel();
-extern void AddSC_boss_deathbringer_saurfang();
-extern void AddSC_boss_festergut();
-extern void AddSC_boss_lady_deathwhisper();
-extern void AddSC_boss_lord_marrowgar();
-extern void AddSC_boss_professor_putricide();
-extern void AddSC_boss_rotface();
-extern void AddSC_boss_sindragosa();
-extern void AddSC_boss_the_lich_king();
-extern void AddSC_boss_valithria_dreamwalker();
-extern void AddSC_gunship_battle();
-extern void AddSC_instance_icecrown_citadel();
-extern void AddSC_boss_anubrekhan();                        // naxxramas
-extern void AddSC_boss_four_horsemen();
-extern void AddSC_boss_faerlina();
-extern void AddSC_boss_gluth();
-extern void AddSC_boss_gothik();
-extern void AddSC_boss_grobbulus();
-extern void AddSC_boss_kelthuzad();
-extern void AddSC_boss_loatheb();
-extern void AddSC_boss_maexxna();
-extern void AddSC_boss_noth();
-extern void AddSC_boss_heigan();
-extern void AddSC_boss_patchwerk();
-extern void AddSC_boss_razuvious();
-extern void AddSC_boss_sapphiron();
-extern void AddSC_boss_thaddius();
-extern void AddSC_instance_naxxramas();
-extern void AddSC_boss_malygos();                           // nexus, eye_of_eternity
-extern void AddSC_boss_anomalus();                          // nexus, nexus
-extern void AddSC_boss_keristrasza();
-extern void AddSC_boss_ormorok();
-extern void AddSC_boss_telestra();
-extern void AddSC_instance_nexus();
-extern void AddSC_boss_sartharion();                        // obsidian_sanctum
-extern void AddSC_instance_obsidian_sanctum();
-extern void AddSC_boss_baltharus();                         // ruby_sanctum
-extern void AddSC_boss_halion();
-extern void AddSC_boss_saviana();
-extern void AddSC_boss_zarithrian();
-extern void AddSC_instance_ruby_sanctum();
-extern void AddSC_boss_bjarngrim();                         // ulduar, halls_of_lightning
-extern void AddSC_boss_ionar();
-extern void AddSC_boss_loken();
-extern void AddSC_boss_volkhan();
-extern void AddSC_instance_halls_of_lightning();
-extern void AddSC_boss_maiden_of_grief();                   // ulduar, halls_of_stone
-extern void AddSC_boss_sjonnir();
-extern void AddSC_halls_of_stone();
-extern void AddSC_instance_halls_of_stone();
-extern void AddSC_boss_assembly_of_iron();                  // ulduar, ulduar
-extern void AddSC_boss_algalon();
-extern void AddSC_boss_auriaya();
-extern void AddSC_boss_flame_leviathan();
-extern void AddSC_boss_freya();
-extern void AddSC_boss_general_vezax();
-extern void AddSC_boss_hodir();
-extern void AddSC_boss_ignis();
-extern void AddSC_boss_kologarn();
-extern void AddSC_boss_mimiron();
-extern void AddSC_boss_razorscale();
-extern void AddSC_boss_thorim();
-extern void AddSC_boss_xt_002();
-extern void AddSC_boss_yogg_saron();
-extern void AddSC_instance_ulduar();
-extern void AddSC_ulduar();
-extern void AddSC_boss_ingvar();                            // utgarde_keep, utgarde_keep
-extern void AddSC_boss_keleseth();
-extern void AddSC_boss_skarvald_and_dalronn();
-extern void AddSC_instance_utgarde_keep();
-extern void AddSC_utgarde_keep();
-extern void AddSC_boss_gortok();                           // utgarde_keep, utgarde_pinnacle
-extern void AddSC_boss_skadi();
-extern void AddSC_boss_svala();
-extern void AddSC_boss_ymiron();
-extern void AddSC_instance_pinnacle();
-extern void AddSC_boss_archavon();                          // vault_of_archavon
-extern void AddSC_boss_emalon();
-extern void AddSC_boss_koralon();
-extern void AddSC_boss_toravon();
-extern void AddSC_instance_vault_of_archavon();
-extern void AddSC_boss_erekem();                            // violet_hold
-extern void AddSC_boss_ichoron();
-extern void AddSC_instance_violet_hold();
-extern void AddSC_violet_hold();
+// removed
 
-extern void AddSC_borean_tundra();
-extern void AddSC_dalaran();
-extern void AddSC_dragonblight();
-extern void AddSC_grizzly_hills();
-extern void AddSC_howling_fjord();
-extern void AddSC_icecrown();
-extern void AddSC_sholazar_basin();
-extern void AddSC_storm_peaks();
-extern void AddSC_zuldrak();
-
 // outland
 extern void AddSC_boss_exarch_maladaar();                   // auchindoun, auchenai_crypts
 extern void AddSC_boss_shirrak();
@@ -598,7 +474,22 @@
     AddSC_boss_ragnaros();
     AddSC_instance_molten_core();
     AddSC_molten_core();
-    AddSC_ebon_hold();                                      // scarlet_enclave
+    AddSC_boss_anubrekhan();                                // naxxramas
+    AddSC_boss_four_horsemen();
+    AddSC_boss_faerlina();
+    AddSC_boss_gluth();
+    AddSC_boss_gothik();
+    AddSC_boss_grobbulus();
+    AddSC_boss_kelthuzad();
+    AddSC_boss_loatheb();
+    AddSC_boss_maexxna();
+    AddSC_boss_noth();
+    AddSC_boss_heigan();
+    AddSC_boss_patchwerk();
+    AddSC_boss_razuvious();
+    AddSC_boss_sapphiron();
+    AddSC_boss_thaddius();
+    AddSC_instance_naxxramas();
     AddSC_boss_arcanist_doan();                             // scarlet_monastery
     AddSC_boss_azshir_the_sleepless();
     AddSC_boss_bloodmage_thalnos();
@@ -717,8 +608,6 @@
     AddSC_boss_lieutenant_drake();
     AddSC_instance_old_hillsbrad();
     AddSC_old_hillsbrad();
-    AddSC_culling_of_stratholme();                          // CoT, culling_of_stratholme
-    AddSC_instance_culling_of_stratholme();
     AddSC_dire_maul();                                      // dire_maul
     AddSC_instance_dire_maul();
     AddSC_boss_celebras_the_cursed();                       // maraudon
@@ -773,145 +662,8 @@
     AddSC_winterspring();
 
     // northrend
-    AddSC_boss_jedoga();                                    // azjol-nerub, ahnkahet
-    AddSC_boss_nadox();
-    AddSC_boss_taldaram();
-    AddSC_boss_volazj();
-    AddSC_instance_ahnkahet();
-    AddSC_boss_anubarak();                                  // azjol-nerub, azjol-nerub
-    AddSC_boss_hadronox();
-    AddSC_boss_krikthir();
-    AddSC_instance_azjol_nerub();
-    AddSC_boss_grand_champions();                           // CC, trial_of_the_champion
-    AddSC_instance_trial_of_the_champion();
-    AddSC_trial_of_the_champion();
-    AddSC_boss_anubarak_trial();                            // CC, trial_of_the_crusader
-    AddSC_boss_faction_champions();
-    AddSC_boss_jaraxxus();
-    AddSC_instance_trial_of_the_crusader();
-    AddSC_northrend_beasts();
-    AddSC_trial_of_the_crusader();
-    AddSC_twin_valkyr();
-    AddSC_boss_novos();                                     // draktharon_keep
-    AddSC_boss_tharonja();
-    AddSC_boss_trollgore();
-    AddSC_instance_draktharon_keep();
-    AddSC_boss_colossus();                                  // gundrak
-    AddSC_boss_eck();
-    AddSC_boss_galdarah();
-    AddSC_boss_moorabi();
-    AddSC_boss_sladran();
-    AddSC_instance_gundrak();
-    AddSC_boss_bronjahm();                                  // ICC, FH, forge_of_souls
-    AddSC_boss_devourer_of_souls();
-    AddSC_instance_forge_of_souls();
-    AddSC_boss_falric();                                    // ICC, FH, halls_of_reflection
-    AddSC_boss_lich_king();
-    AddSC_boss_marwyn();
-    AddSC_halls_of_reflection();
-    AddSC_instance_halls_of_reflection();
-    AddSC_boss_gafrost();                                   // ICC, FH, pit_of_saron
-    AddSC_boss_krick_and_ick();
-    AddSC_boss_tyrannus();
-    AddSC_instance_pit_of_saron();
-    AddSC_pit_of_saron();
-    AddSC_blood_prince_council();                           // ICC, icecrown_citadel
-    AddSC_boss_blood_queen_lanathel();
-    AddSC_boss_deathbringer_saurfang();
-    AddSC_boss_festergut();
-    AddSC_boss_lady_deathwhisper();
-    AddSC_boss_lord_marrowgar();
-    AddSC_boss_professor_putricide();
-    AddSC_boss_rotface();
-    AddSC_boss_sindragosa();
-    AddSC_boss_the_lich_king();
-    AddSC_boss_valithria_dreamwalker();
-    AddSC_gunship_battle();
-    AddSC_instance_icecrown_citadel();
-    AddSC_boss_anubrekhan();                                // naxxramas
-    AddSC_boss_four_horsemen();
-    AddSC_boss_faerlina();
-    AddSC_boss_gluth();
-    AddSC_boss_gothik();
-    AddSC_boss_grobbulus();
-    AddSC_boss_kelthuzad();
-    AddSC_boss_loatheb();
-    AddSC_boss_maexxna();
-    AddSC_boss_noth();
-    AddSC_boss_heigan();
-    AddSC_boss_patchwerk();
-    AddSC_boss_razuvious();
-    AddSC_boss_sapphiron();
-    AddSC_boss_thaddius();
-    AddSC_instance_naxxramas();
-    AddSC_boss_malygos();                                   // nexus, eye_of_eternity
-    AddSC_boss_anomalus();                                  // nexus, nexus
-    AddSC_boss_keristrasza();
-    AddSC_boss_ormorok();
-    AddSC_boss_telestra();
-    AddSC_instance_nexus();
-    AddSC_boss_sartharion();                                // obsidian_sanctum
-    AddSC_instance_obsidian_sanctum();
-    AddSC_boss_baltharus();                                 // ruby_sanctum
-    AddSC_boss_halion();
-    AddSC_boss_saviana();
-    AddSC_boss_zarithrian();
-    AddSC_instance_ruby_sanctum();
-    AddSC_boss_bjarngrim();                                 // ulduar, halls_of_lightning
-    AddSC_boss_ionar();
-    AddSC_boss_loken();
-    AddSC_boss_volkhan();
-    AddSC_instance_halls_of_lightning();
-    AddSC_boss_maiden_of_grief();                           // ulduar, halls_of_stone
-    AddSC_boss_sjonnir();
-    AddSC_halls_of_stone();
-    AddSC_instance_halls_of_stone();
-    AddSC_boss_assembly_of_iron();                          // ulduar, ulduar
-    AddSC_boss_algalon();
-    AddSC_boss_auriaya();
-    AddSC_boss_flame_leviathan();
-    AddSC_boss_freya();
-    AddSC_boss_general_vezax();
-    AddSC_boss_hodir();
-    AddSC_boss_ignis();
-    AddSC_boss_kologarn();
-    AddSC_boss_mimiron();
-    AddSC_boss_razorscale();
-    AddSC_boss_thorim();
-    AddSC_boss_xt_002();
-    AddSC_boss_yogg_saron();
-    AddSC_instance_ulduar();
-    AddSC_ulduar();
-    AddSC_boss_ingvar();                                    // UK, utgarde_keep
-    AddSC_boss_keleseth();
-    AddSC_boss_skarvald_and_dalronn();
-    AddSC_instance_utgarde_keep();
-    AddSC_utgarde_keep();
-    AddSC_boss_gortok();                                    // UK, utgarde_pinnacle
-    AddSC_boss_skadi();
-    AddSC_boss_svala();
-    AddSC_boss_ymiron();
-    AddSC_instance_pinnacle();
-    AddSC_boss_archavon();                                  // vault_of_archavon
-    AddSC_boss_emalon();
-    AddSC_boss_koralon();
-    AddSC_boss_toravon();
-    AddSC_instance_vault_of_archavon();
-    AddSC_boss_erekem();                                    // violet_hold
-    AddSC_boss_ichoron();
-    AddSC_instance_violet_hold();
-    AddSC_violet_hold();
+    // removed
 
-    AddSC_borean_tundra();
-    AddSC_dalaran();
-    AddSC_dragonblight();
-    AddSC_grizzly_hills();
-    AddSC_howling_fjord();
-    AddSC_icecrown();
-    AddSC_sholazar_basin();
-    AddSC_storm_peaks();
-    AddSC_zuldrak();
-
     // outland
     AddSC_boss_exarch_maladaar();                           // auchindoun, auchenai_crypts
     AddSC_boss_shirrak();
